### DISCUSSION TITLE
Issue with Material-UI Tabs (dynamic field names)

### DISCUSSION THREAD

**Discussion by mbretter** [react-hook-form/react-hook-form#5771] [Q&A]
**Describe the bug**
When adding dynamic field names like `lang.${lang}.name` using a Controller and a material-ui Textfield, the values are not displayed correctly when switching over the tab/language.

**To Reproduce**

1. Go to the code sandbox
2. English is displayed in the textfield
3. Switch to DE
4. English ist still displayed, but Deutsch should appear

**Codesandbox link (Required)**
https://codesandbox.io/s/rhf-basic-muitabs-icp7m

**Expected behavior**
When switching the tabs, the value in the textfield should reflect the field value for the corresponding language.

**Desktop (please complete the following information):**

- OS: linux
- Browser chrome/firefox
- Version 87/89

**Additional context**
When pressing "console log form state" after startup, the log shows
```
{lang: {en: {name: "English"}}}
```
which is ok so far, but after switching (without typing anything)
```
{lang: {de: {name: "English"}}}
```
this is wrong, it looks like that the value got overwritten.

**Comment by bluebill1049** [comment]
have a look at this section: https://react-hook-form.com/faqs#Whyisdefaultvaluenotchangingcorrectlywithternaryoperator give a different key should resolve this issue.

**Reply by mbretter** [reply]
I have updated the csb and added a key prop to the Controller.
When switching to DE, Deutsch is displayed (this is ok), when switching back, English is not display anymore, and the console log of the form state shows undefined as value.

**Reply by bluebill1049** [reply]
that's because inputs get unmounted and `defaultValues` gets removed as well. same behaviour when you unmount an input.

**Reply by mbretter** [reply]
hmm, this behavior looks new to me, I am sure, that this already worked.
is there any option to keep the default values on unmount?

**Reply by bluebill1049** [reply]
unmount input, basically invoke `unregister`,  `unregister` by default, remove all `formState` (include defaultValue). 

<img width="936" alt="Screen Shot 2021-06-29 at 8 09 06 pm" src="https://user-images.githubusercontent.com/10513364/123779763-dcfc6080-d915-11eb-9bf4-a0f41d72abb5.png">

there is no option at the moment to keep `formState` with `shouldUnregister` set to true, beside of reset the form when you switch between different tabs, or use useForm for each tab.

**Reply by mbretter** [reply]
when switching back to 7.8.2, everything is working fine, this issue must be introduced with 7.9.0, seems to be a regression?
The problem when setting shouldUnregister to false is, that all default values will be submitted, even those with no related form field.

**Reply by mbretter** [reply]
figured it out, 7.8.3 is the last working version, starting with 7.8.4 the problem happens

**Reply by bluebill1049** [reply]
good one! you are right.
